app/views/doorkeeper/applications/_form.html.erb

```diff
@@ -1,4 +1,4 @@
-<%= form_for :application, url: oauth_applications_path(application), html: {class: 'form-horizontal', role: 'form'} do |f| %>
+<%= form_for :application, url: application.new_record? ? oauth_applications_path(application) : oauth_application_path(application), method: application.new_record? ? :post : :put, html: {class: 'form-horizontal', role: 'form'} do |f| %>
```

Comment:
can this be like the doorkeeper form instead? https://github.com/doorkeeper-gem/doorkeeper/blob/master/app/views/doorkeeper/applications/_form.html.erb#L1